hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -371,9 +365,9 @@ public HoodieTableConfig(HoodieStorage storage, StoragePath metaPath, String pay
         setValue(PAYLOAD_CLASS_NAME, payloadClassName);
         needStore = true;
       }
-      if (contains(PAYLOAD_TYPE) && payloadClassName != null
-          && !payloadClassName.equals(RecordPayloadType.valueOf(getString(PAYLOAD_TYPE)).getClassName())) {
-        setValue(PAYLOAD_TYPE, RecordPayloadType.fromClassName(payloadClassName).name());
+      if (contains(RECORD_MERGE_MODE) && recordMergeMode != null
```

Comment:
Should we throw an error here since it's going to change the record merge mode? Similar for the payload class as we are not going to allow payload class to change in the table configs (unless through a custom tooling).

Reply:
Is this already handled somewhere?